server/controllers/signup.js

```diff
@@ -0,0 +1,35 @@
+const User = require('../models/UserModel');
+const { genJWT } = require('../utility/util');
+
+const handleRegister = (req, res, bcrypt) => {
+  const MINLENGTH = 6;
+  const SALT = 10;
+  const { email, password, name, confirmPassword } = req.body;
+  if (!email || !name || !password || !confirmPassword) {
+    return res.status(400).json('All fields must be filled');
+  }
+  if (confirmPassword !== password) {
+    return res.status(400).json('Passwords do not match');
+  }
+  if (password.length < MINLENGTH) {
+    return res.status(400).json(`Password must be at least ${MINLENGTH}`);
+  }
+  const hash = bcrypt.hashSync(password, SALT);
```

Comment:
If the hash is performed here, then this code will also need to be added in the route where a password can be updated. It would be preferable to have this be performed automatically when the password is updated or added. Check out the Mongoose pre save functionality, that should allow you to guarantee the password is hashed

Reply:
moved hashing to UserModel.js using mongoose pre hook. Also moved the password comparison to user schema method, to completely remove direct bcrypt dependency in login and signup controller. Validation of password and hashing for new users is now done in the UserModel.js as a middleware